pkg/tsdb/postgres/macros.go

```diff
@@ -79,11 +79,11 @@ func (m *PostgresMacroEngine) evaluateMacro(name string, args []string) (string,
 		}
 		return fmt.Sprintf("extract(epoch from %s) as \"time\"", args[0]), nil
 	case "__timeFilter":
-		// don't use to_timestamp in this macro for redshift compatibility #9566
 		if len(args) == 0 {
 			return "", fmt.Errorf("missing time column argument for macro %v", name)
 		}
-		return fmt.Sprintf("extract(epoch from %s) BETWEEN %d AND %d", args[0], m.TimeRange.GetFromAsSecondsEpoch(), m.TimeRange.GetToAsSecondsEpoch()), nil
+
+		return fmt.Sprintf("%s BETWEEN '%s' AND '%s'", args[0], m.TimeRange.GetFromAsTimeUTC().Format(time.RFC3339), m.TimeRange.GetToAsTimeUTC().Format(time.RFC3339)), nil
 	case "__timeFrom":
```

Comment:
For consistency I think that `$__timeFrom` and `$__timeTo` should return a value formatted as`'2006-01-02T15:04:05Z07:00'` like `$__timeFilter` instead of `to_timestamp(<epoch>)`. What do you think?

Reply:
I agree `$__timeFrom` and `$__timeTo` should be adjusted. If someone really needs the old functionality you can still get it with the unix timestamp macros and pass those into to_timestamp but they should evaluate to the same thing anyway.